common/transport/amqp/devdoc/amqp_requirements.md

```diff
@@ -92,14 +92,14 @@ Configures a `ReceiverLink` object to use the endpoint passed as a parameter and
 
 ### attachSenderLink(endpoint, properties, done)
 **SRS_NODE_COMMON_AMQP_16_012: [** The `attachSenderLink` method shall throw a ReferenceError if the `endpoint` argument is falsy. **]**
-**SRS_NODE_COMMON_AMQP_16_013: [** The `attachSenderLink` method shall call `createSender` on the `amqp10` client object. **]**
+**SRS_NODE_COMMON_AMQP_16_013: [** The `attachSenderLink` method shall call `open_sender` on the `rhea` session object. **]**
 **SRS_NODE_COMMON_AMQP_06_003: [** The `attachSenderLink` method shall create a policy object that contain link options to be merged if the linkOptions argument is not falsy. **]**
 **SRS_NODE_COMMON_AMQP_16_015: [** The `attachSenderLink` method shall call the `done` callback with a `null` error and the link object that was created if the link was attached successfully. **]**
 **SRS_NODE_COMMON_AMQP_16_016: [** The `attachSenderLink` method shall call the `done` callback with an `Error` object if the link object wasn't created successfully. **]**
 
 ### attachReceiverLink(endpoint, properties, done)
 **SRS_NODE_COMMON_AMQP_16_017: [** The `attachReceiverLink` method shall throw a ReferenceError if the `endpoint` argument is falsy. **]**
-**SRS_NODE_COMMON_AMQP_16_018: [** The `attachReceiverLink` method shall call `createReceiver` on the `amqp10` client object. **]**
+**SRS_NODE_COMMON_AMQP_16_018: [** The `attachReceiverLink` method shall call `open_receiver` on the `rhea` session object. **]**
```

Comment:
>session [](start = 108, length = 7)

how is that session object obtained?

Reply:
New state between connecting and connected.  Called connecting_session.  sync call to create_session followed by open, with the "standard" handler setup.



---
In reply to: [191906146](https://github.com/Azure/azure-iot-sdk-node/pull/300#discussion_r191906146) [](ancestors = 191906146)